modules/tensor_mechanics/src/materials/HillPlasticityStressUpdate.C

```diff
@@ -125,71 +139,78 @@ ADHillPlasticityStressUpdate::computeDeltaDerivatives(const ADReal & delta_gamma
 
-ADReal
-ADHillPlasticityStressUpdate::computeResidual(const ADDenseVector & stress_dev,
-                                              const ADDenseVector & /*stress_sigma*/,
-                                              const ADReal & delta_gamma)
+template <bool is_ad>
+GenericReal<is_ad>
+HillPlasticityStressUpdateTempl<is_ad>::computeResidual(
+    const GenericDenseVector<is_ad> & stress_dev,
+    const GenericDenseVector<is_ad> & /*stress_sigma*/,
+    const GenericReal<is_ad> & delta_gamma)
 {
 
   // If in elastic regime, just return
   if (_yield_condition <= 0.0)
     return 0.0;
 
-  ADDenseMatrix eigenvectors_hill_transpose(6, 6);
+  GenericDenseMatrix<is_ad> eigenvectors_hill_transpose(6, 6);
 
   _eigenvectors_hill.get_transpose(eigenvectors_hill_transpose);
   eigenvectors_hill_transpose.vector_mult(_stress_np1, stress_dev);
 
-  ADReal omega = computeOmega(delta_gamma, _stress_np1);
+  GenericReal<is_ad> omega = computeOmega(delta_gamma, _stress_np1);
   _hardening_slope = computeHardeningDerivative();
 
   // Hardening variable is \alpha isotropic hardening for now.
   _hardening_variable[_qp] = computeHardeningValue(delta_gamma, omega);
-  ADReal s_y = _hardening_slope * _hardening_variable[_qp] + _yield_stress;
+  GenericReal<is_ad> s_y = _hardening_slope * _hardening_variable[_qp] + _yield_stress;
 
-  ADReal residual = 0.0;
+  GenericReal<is_ad> residual = 0.0;
   residual = s_y / omega - 1.0;
 
   return residual;
 }
 
-ADReal
-ADHillPlasticityStressUpdate::computeDerivative(const ADDenseVector & /*stress_dev*/,
-                                                const ADDenseVector & /*stress_sigma*/,
-                                                const ADReal & delta_gamma)
+template <bool is_ad>
+GenericReal<is_ad>
+HillPlasticityStressUpdateTempl<is_ad>::computeDerivative(
+    const GenericDenseVector<is_ad> & /*stress_dev*/,
+    const GenericDenseVector<is_ad> & /*stress_sigma*/,
+    const GenericReal<is_ad> & delta_gamma)
 {
   // If in elastic regime, return unit derivative
   if (_yield_condition <= 0.0)
     return 1.0;
 
-  ADReal omega = computeOmega(delta_gamma, _stress_np1);
+  GenericReal<is_ad> omega = computeOmega(delta_gamma, _stress_np1);
   _hardening_slope = computeHardeningDerivative();
 
-  ADReal sy = _hardening_slope * computeHardeningValue(delta_gamma, omega) + _yield_stress;
-  ADReal sy_alpha = _hardening_slope;
+  GenericReal<is_ad> sy =
+      _hardening_slope * computeHardeningValue(delta_gamma, omega) + _yield_stress;
+  GenericReal<is_ad> sy_alpha = _hardening_slope;
 
-  ADReal omega_gamma;
-  ADReal sy_gamma;
+  GenericReal<is_ad> omega_gamma;
+  GenericReal<is_ad> sy_gamma;
 
   computeDeltaDerivatives(delta_gamma, _stress_np1, sy_alpha, omega, omega_gamma, sy_gamma);
-  ADReal residual_derivative = 1 / omega * (sy_gamma - 1 / omega * omega_gamma * sy);
+  GenericReal<is_ad> residual_derivative = 1 / omega * (sy_gamma - 1 / omega * omega_gamma * sy);
 
   return residual_derivative;
```

Comment:
Okay, I changed the return type to `Real`.

Reply:
Actually it seems that downstream we rely on a `Generic` type for that method. Since it was already `ADReal`, I think it's fine how you had it... Sorry for the bad suggestion. I still think it does not need to be AD though.

Reply:
Changed it back to Generic.